### DISCUSSION TITLE
Типичная организация, ее структура и основные связи

### DISCUSSION THREAD

**Discussion by vagr-ant** [fhir-ru/core#58] [Административная РГ]
Давайте попытаемся изложить потребности в реализации логической и компонентной структуры типичной медицинской организации.
Попытаюсь описать "сложный случай" исходя из своего опыта разработки. 
Комментарии приветствуются. Для начала хотелось бы собрать "вводные", а уж потом переходить к реализации.

(i) Базовое описание для начала дискуссии (i)

### Вводные
Имеем сетевую клинику ООО "Рога и копыта" (далее _**организация**_) имеющую в своем составе два отдельных Юридических лица:

- ООО "Рога"

- ООО "Копыта"

Необходимость двух юридических лиц может быть совершенно разная, но в общем случае у каждого ЮЛ есть:

1. Юридический адрес
2. Банковские реквизиты (Множество)
3. Лицензии на мед. деятельность (Множество, сроки действия, история - тема для отдельной дискуссии)
4. Контакты
5. ИНН, КПП, ОКВЭД и прочие атрибуты.
6. ... будем дополнять

Назовем такое ЮЛ **_подразделением_**.
У каждого **_подразделения_** может быть множество _**отделений**_ (травмотология, хирургия, гинекология и т.п.)
Имеется общий список сотрудников (физических лиц), работающий в _**организации**_
Каждый сотрудник может быть привязан к отделению с определенной ролью. Один и тот же сотрудник может работать в разных отделениях с разной ролью. Для каждого "места работы" свой режим/расписание.

У _**организации**_ имеется несколько физических площадок (зданий).
Внутри зданий располагаются отделения, причем в одном здании могут располагаться отделения разных ЮЛ.
Например: "Рога" >> "травматология" и "Копыта" >> "детская травматология" (потому что у "Копыт" есть детская лицензия, а у "Рогов" нет)
Важно! Лицензия на мед. деятельность имеет привязку к ЮЛ на площадке выполнения (необходимо учесть)

В пределах одной физической площадки организация не может оказывать одну и ту же услугу с разной стоимостью (есть понятие скидок/надбавок и т.п., но в данном случае речь идет про Прайс), но на разных площадках он может быть разным (vip-отделение, обычное и т.п.) - появится "Прайс-лист по площадке выполнения"

Помним про запись на прием к специалисту (адрес, режим работы и т.д.)

### Вопросы
- Что считаем за базовую единицу organization и как связываем друг с другом исходя из вводных?

- Для ответа на вопрос выше важно учесть, что PractitionerRole (а видимо он будет отвечать за сотрудника в отделении с ролью)  ссылается на organization

### Предложения по реализации на FHIR
... to be

**Comment by shamilnn** [comment]
На мой взгляд, всё вышеперечисленное тема для отдельного IG под условным названием Provider Registry, поверх или на основе RU Core.
![registry](https://user-images.githubusercontent.com/75683852/144710392-783cc1ec-49e7-4668-b74f-2dbda401c62a.png)

**Comment by Vrodvi** [comment]
Можно взять реально существующую организацию, например "ГБУЗ МКНЦ имени А.С. Логинова ДЗМ" -  https://github.com/fhir-ru/core/discussions/26#discussioncomment-1693435, структура там достаточно сложная.
